<?xml version="1.0"?>
<!--
    Default odometry launch file
-->

<launch>

    <!-- Launch file arguments -->
    <arg name="start_serial" default="true" />
    <arg name="start_actuation_reader" default="true" />
    <arg name="vehicle_name"  default="" />
    <arg name="encoder_frame" default="base_link" />
    <arg name="encoder_topic" default="lli/encoder" />
    <arg name="direction_topic" default="actuation_twist" />


    <!-- Velocity estimation -->
    <group if="$(arg start_serial)">
        <node pkg="svea_sensors" type="actuation_to_twist.py" output="screen" name="actuation_to_twist" />
    </group>

    <!-- Read from encoder -->
    <node pkg="svea_sensors" type="wheel_encoder_reader.py" output="screen" name="wheel_encoder_reader">
        <param name="vehicle_name" value="$(arg vehicle_name)" />
        <param name="encoder_frame" value="$(arg encoder_frame)" />
        <param name="encoder_topic" value="$(arg encoder_topic)" />
        <param name="direction_topic" value="$(arg direction_topic)" />
    </node>

    <!-- Start low-level interface -->
    <group if="$(arg start_serial)">
        <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="250000" />
        </node>
    </group>

</launch>
